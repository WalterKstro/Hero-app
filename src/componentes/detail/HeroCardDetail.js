const HeroCardDetail = ({
        characters,
        first_appearance,
        id,
        publisher,
        superhero,
        back,
        length,
        push
}) => {
    const handlerToBack = length => {
        if(length <= 2) {
            push('/')
        }else {
            back()
        }
    }
    return (
       <div className="row">
           <div className="col-12">
           <div className="d-flex justify-content-center animate__animated animate__zoomIn" style={{maxWidth: 100 + '%'}}>
            <div className="card mt-5" style={{maxWidth: 100 + '%'}}>
                <div className="row g-0">
                    <div className="col-md-4">
                    <img src={`${process.env.PUBLIC_URL}/assets/heroes/${id}.jpg`} className="img-fluid rounded-start" alt="..." />
                    </div>
                    <div className="col-md-8">
                        <div className="card-body">
                            <h3 className="card-title my-3 text-center">{superhero}</h3>
                            <ul className="list-group mb-3">
                                <li className="list-group-item d-flex justify-content-between align-items-start">
                                    <div className="ms-2 me-auto">
                                        <div className="fw-bold">Company</div>
                                        {publisher}
                                    </div>
                                </li>
                                <li className="list-group-item d-flex justify-content-between align-items-start">
                                    <div className="ms-2 me-auto">
                                        <div className="fw-bold">Name</div>
                                        {characters}
                                    </div>
                                </li>
                                <li className="list-group-item d-flex justify-content-between align-items-start">
                                    <div className="ms-2 me-auto">
                                        <div className="fw-bold">First Appearance</div>
                                        {first_appearance}
                                    </div>
                                </li>
                            </ul>
                        <button 
                            className="btn btn-outline-primary"
                            onClick={ () => handlerToBack(length) }
                            >Go Back</button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
           </div>
       </div>
    )
}

export default HeroCardDetail
