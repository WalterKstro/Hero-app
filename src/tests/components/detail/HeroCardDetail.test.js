import { shallow } from "enzyme"
import HeroCardDetail from "../../../componentes/detail/HeroCardDetail"

describe('Testing <HeroCardDetail/>', () => {

    let props = undefined
    let rendered = undefined
    
    beforeEach(() => {
        props = {
            alter_ego: "Bruce Wayne",
            characters: "Bruce Wayne",
            first_appearance: "Detective Comics #27",
            id: "dc-batman",
            publisher: "DC Comics",
            superhero: "Batman",
            back : jest.fn(),
            length : 1,
            push : jest.fn()
        }

        rendered = shallow(
            <HeroCardDetail {...props} />
        )
    })

    test('should show correctly', () => {
        expect(rendered).toMatchSnapshot()
    })


    test('should return to home page with the method push()', () => {

        rendered.find('.btn-outline-primary').simulate('click')
        expect(props.push).toHaveBeenCalledWith('/')
        expect(props.back).not.toHaveBeenCalled()

    })

    test('should return to back with the method back()', () => {

        props = {...props, length : 10}
        const rendered = shallow(
            <HeroCardDetail {...props} />
        )

        rendered.find('.btn-outline-primary').simulate('click')

        expect(props.back).toBeCalledTimes(1)
        expect(props.back).toBeCalledWith()
        expect(props.push).not.toHaveBeenCalled()
        
    })
    
    
})
