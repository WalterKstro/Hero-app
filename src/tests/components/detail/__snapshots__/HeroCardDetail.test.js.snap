// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HeroCardDetail/> should show correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <div
      className="d-flex justify-content-center animate__animated animate__zoomIn"
      style={
        Object {
          "maxWidth": "100%",
        }
      }
    >
      <div
        className="card mt-5"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      >
        <div
          className="row g-0"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="..."
              className="img-fluid rounded-start"
              src="/assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h3
                className="card-title my-3 text-center"
              >
                Batman
              </h3>
              <ul
                className="list-group mb-3"
              >
                <li
                  className="list-group-item d-flex justify-content-between align-items-start"
                >
                  <div
                    className="ms-2 me-auto"
                  >
                    <div
                      className="fw-bold"
                    >
                      Company
                    </div>
                    DC Comics
                  </div>
                </li>
                <li
                  className="list-group-item d-flex justify-content-between align-items-start"
                >
                  <div
                    className="ms-2 me-auto"
                  >
                    <div
                      className="fw-bold"
                    >
                      Name
                    </div>
                    Bruce Wayne
                  </div>
                </li>
                <li
                  className="list-group-item d-flex justify-content-between align-items-start"
                >
                  <div
                    className="ms-2 me-auto"
                  >
                    <div
                      className="fw-bold"
                    >
                      First Appearance
                    </div>
                    Detective Comics #27
                  </div>
                </li>
              </ul>
              <button
                className="btn btn-outline-primary"
                onClick={[Function]}
              >
                Go Back
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
